android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskExpireUtil.kt

```diff
@@ -70,7 +70,9 @@ constructor(@ApplicationContext val appContext: Context, val fhirEngine: FhirEng
           count = tasksCount
           sort(Task.AUTHORED_ON, Order.ASCENDING)
         }
+        .asSequence()
         .filter { it.isPastExpiry() }
+        .toList()
```

Comment:
Here we have a stateful operation happening on each item of a list, so using a sequence here is not ideal. So converting this to a list is not required. See [Working with stateless/stateful operations](https://typealias.com/guides/when-to-use-sequences/)

Reply:
Maybe we can discuss more about this on a call or huddle.
Here's a sample 
``` kotlin  
val result = collectionOf12Items
    .asSequence()
    .map { it.toString() }
    .toList() 
```